### ISSUE TITLE
False positives: emphasis lines flagged as list markers (MD030) and list-adjacent code fences (MD032)

### ISSUE THREAD

**Issue by juherr** [joshrotenberg/mdbook-lint#438]
### Summary

Two standard-rule checks fire false positives on CommonMark-valid content, because the parser treats constructs that are **not** lists as lists:

- **MD030** flags a line that starts with `*<text>` (emphasis) as an unordered-list marker.
- **MD032** flags an indented code fence that belongs to a list item as if the list ended before it.

Neither is reported by `markdownlint-cli2` (which uses a CommonMark-compliant parser), so this looks like a parsing deviation rather than a rules-definition difference.

### Environment

- mdbook-lint: **0.14.4** (installed from the release binary)
- Invoked as a standalone CLI: `mdbook-lint lint <file>`

---

### Bug 1 — MD030 false positive on emphasis at line start

A line beginning with `*` immediately followed by a non-space character is **emphasis**, not a list item (a list marker requires `* ` — marker + space). mdbook-lint treats it as an unordered-list marker with zero spaces.

**Reproduction** (`emphasis.md`):

```markdown
# Title

*Rejected — socket peer address (`ConnectInfo`).* Non-spoofable, but behind Cloudflare/Render every
external client collapses onto the proxy IP, so the per-client limiter is meaningless.

*Rejected — rightmost `X-Forwarded-For` with N trusted hops.* Generic, but calibrating the hop count
across two proxies is fragile and error-prone; `CF-Connecting-IP` is exact for this topology.

*Rejected — per-interface pre-auth limiters (one per nest).* Gives interface-correct error bodies
natively, but splits the IP budget across `/api` and `/ocpi`.
```

```console
$ mdbook-lint lint --enable MD030,MD032 emphasis.md
warning[MD030]: Unordered list marker spacing: expected 1 space(s) after '*', found 0
  --> emphasis.md:6:2
warning[MD030]: Unordered list marker spacing: expected 1 space(s) after '*', found 0
  --> emphasis.md:9:2
```

**Expected:** no MD030 warning — these are emphasis spans, not list items.

Notes:

- The warning only appears when a list-detecting rule is also active (e.g. `--enable MD030,MD032`, or the default rule set); `--enable MD030` alone reports nothing. This points at shared list parsing as the source.
- The **first** emphasis paragraph is not flagged, but subsequent ones are — as if the first line opens a phantom list and the following `*…` lines are parsed as its items.

`markdownlint-cli2@0.23.1` on the same file reports only `MD013` (line length); never `MD030`.

---

### Bug 2 — MD032 false positive on an indented code fence inside a list item

An indented fenced code block that is part of a list item is flagged as if the list needed a blank line before the fence.

**Reproduction** (`listcode.md`):

````markdown
# Title

- Inspect applied migrations:

  ```sql
  select version from _sqlx_migrations;
  ```
````

```console
$ mdbook-lint lint --enable MD032 listcode.md
warning[MD032]: List should be followed by a blank line
  --> listcode.md:5:1
```

**Expected:** no MD032 warning — the indented fence is a continuation of the list item, so the list has not ended.

`markdownlint-cli2@0.23.1` on the same file reports **0 issues**.

---

### Impact

Both fire on ordinary documentation (ADRs with a "Considered alternatives" section written in emphasis; runbooks with code blocks inside list items), forcing MD030/MD032 to be disabled to keep CI green — which loses genuine coverage of those rules. Because `markdownlint-cli2` handles both correctly, the fix likely lives in mdbook-lint's list/emphasis parsing rather than the rule logic.

**Comment by joshrotenberg** [comment]
ready: Working this in two slices. Slice 1 (MD030) is in draft PR #446.

Root cause of the MD030 half, and a correction to the diagnosis above: this is not shared list parsing. MD030 is entirely line-based and never touches the AST; `--enable MD030` alone reproduces the warnings on the current build. `MD030::is_emphasis_syntax` accepted a line-leading `*` as emphasis only when the closing `*` fell within the first 50 characters. The first paragraph in the reproduction closes at offset 46 and is accepted; the next two close past 50 and are flagged. That threshold is also why the warning appears to come and go as unrelated text is edited.

PR #446 drops the threshold: a closing `*` anywhere later on the line makes the span emphasis. CommonMark requires a space after a bullet, so `*text*` is never a list item. A line with no closing `*` is still reported as a missing space after a list marker.

Slice 2 (MD032, indented code fence continuing a list item) is not addressed yet and will be a separate PR, so #446 does not close this issue.
